.github/workflows/pytest.yml

```diff
@@ -0,0 +1,45 @@
+name: Unit Tests
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os.host }}
+    strategy:
+      matrix:
+        python-version: ["3.8"]
+        os:
+          - name: ubuntu
+            host: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          auto-update-conda: true
+          python-version: ${{ matrix.python-version }}
+          activate-environment: dap_prinz_green_jobs
+          environment-file: environment.yaml
+      - name: Activate Conda environment
+        run: |
+          eval "$(conda shell.bash hook)"
+          conda activate dap_prinz_green_jobs
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "pip"
+      - name: (ubuntu) Install pip dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -e .
+          pip install pytest
+          python -m spacy download en_core_web_sm
+      - name: (ubuntu) Install conda dependencies
+        run: |
+          conda install -c pytorch faiss-cpu=1.7.4 mkl=2021 blas=1.0=mkl -y
+      - name: set pythonpath
+        run: |
+          echo "PYTHONPATH=dap_prinz_green_jobs" >> $GITHUB_ENV
+      - name: (ubuntu) Test with pytest
+        run: |
+          pytest --verbose
```

Comment:
any ideas welcome! the current error is 'no module named faiss' i think the pythonpath setep is likely not necessary anymore

Reply:
something i did learn is that there are different errors when you test using 3.9 and a windows machine